index.js

```diff
@@ -55,12 +55,12 @@ app.get('/api/users', (req, res) => {
 });
 
 app.get('/api/user/:id', (req, res) => {
-    if (req.isAuthenticated()) {
+    if (req.isAuthenticated() && req.params.id == req.user._id) {
         db.getById('user', req.params.id).then(user => {
             return res.json({ user });
         });
     } else {
-        return res.json({ error: 'Not authenticated' });
```

Comment:
these get caught in the frontend as successful responses for some reason

Reply:
Maybe try `res.status(403).json({error: 'Not authenticated'});`

Reply:
I did right below it and actually was still causing the same behavior.